File: 1797-goal-parser-interpretation/1797-goal-parser-interpretation.cpp
class Solution {
public:
    string interpret(string command) {
        string result = "";
        
        int i = 0;
        int n = command.size();
        while (i < n) {
            if (command[i] == 'G') {
                result += "G";
                i++;
            } else if (command[i] == '(') {
                if (command[i + 1] == ')') {
                    result += "o";
                    i+=2;
                } else if (command[i + 1] == 'a') {
                    result += "al";
                    i+=4;
                }
            }
        }

        return result;
    }
};

File: 1797-goal-parser-interpretation/1797-goal-parser-interpretation.java
class Solution {
    public String interpret(String command) {
        String result = "";

        for (int i = 0; i < command.length(); i++) {
            if (command.charAt(i) == 'G') {
                result += command.charAt(i);
                continue;
            }

            switch(command.charAt(i + 1)) {
                case ')':
                    result += 'o';
                    i++;
                    break;
                case 'a':
                    result += "al";
                    i+=3;
                    break;
            }
        }

        return result;
    }
}